pickhardtpayments/Attempt.py

```diff
@@ -46,26 +56,22 @@ def __init__(self, path: List[UncertaintyChannel], amount: int = 0):
             valid_path = valid_path and (path[i - 1].dest == path[i].src)
             i += 1
         if valid_path:
-            self._path = path
+            self._path: List[Channel] = path
```

Comment:
What happens if the path is not valid? While testing the patch by extending the framework I ran into a case where  the path wasn't valid. but the code broke much later when attempts where attempted to be settled as self._path was not set. I suggest to raise an exception in the else branch.

Reply:
checking path argument and raising a ValueError in d4e38ca